libs/community/langchain_community/document_loaders/parsers/grobid.py

```diff
@@ -83,20 +83,20 @@ def process_xml(
                                 "pages": (fpage, lpage),
                             }
                             chunks.append(sentence_dict)
-                    if segment_sentences is not True:
-                        fpage, lpage = (
-                            chunk_bboxes[0][0]["page"],
-                            chunk_bboxes[-1][-1]["page"],
-                        )
-                        paragraph_dict = {
-                            "text": "".join(paragraph_text),
-                            "para": str(i),
-                            "bboxes": chunk_bboxes,
-                            "section_title": sect.text,
-                            "section_number": sect.get("n"),
-                            "pages": (fpage, lpage),
-                        }
-                        chunks.append(paragraph_dict)
+                        if segment_sentences is not True:
+                            fpage, lpage = (
+                                chunk_bboxes[0][0]["page"],
```

Comment:
seems like you could still get an index error since it's not guaranteed something has been added to chunk_bboxes yet. is the correct fix to add an explicit check that chunk_bboxes isn't empty?

Reply:
The [Grobid docs](https://grobid.readthedocs.io/en/latest/Coordinates-in-PDF/) state, that

> for each entity the coordinates of a structure is provided as a list of bounding boxes, each one separated by a semicolon ```;```, each bounding box being defined by 5 attributes separated by a comma ```,```

So when iteraring a sentence structure, there is an accompanying bounding box, here ```bbox```. ```sbboxes``` should be created and appended to ```chunk_bboxes```.  If you wish to make it extra robust we could check the length of the ```chunk_bbox```.